### PR TITLE
feat: [SRM-13483]: YAML Diff component for internal change source

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots
Screenshot 2023-01-05 at 3 52 54 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by karansaraswat19** [comment]
retrigger typecheck

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/components/EventCards/InternalCSEventCard/InternalCSEventCard.tsx [hunk 1]
```diff
@@ -89,6 +94,57 @@ export default function HarnessFFEventCard({ data }: { data: ChangeEventDTO }):
...
+            {getString('auditTrail.yamlDifference')}
+          </Text>
+          <YAMLDiffView url={internalChangeEvent?.changeEventDetailsLink?.url} />
+        </Container>
+      )}
     </Card>
   )
 }
+
+const YAMLDiffView = ({ url }: { url: string }): JSX.Element => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/components/EventCards/InternalCSEventCard/InternalCSEventCard.tsx:114]: Move component to a new file

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/components/EventCards/InternalCSEventCard/InternalCSEventCard.tsx [hunk 2]
```diff
@@ -89,6 +94,57 @@ export default function HarnessFFEventCard({ data }: { data: ChangeEventDTO }):
...
+    return (
+      <Layout.Horizontal height={140} flex={{ align: 'center-center' }}>
+        <Icon name="steps-spinner" color={Color.GREY_400} size={30} />
+      </Layout.Horizontal>
+    )
+  } else if (error) {
+    return <PageError message={getErrorMessage(error)} onClick={() => refetch()} />
+  } else if (data?.objectsnapshots && data.objectsnapshots.length === 2) {
+    const { objectsnapshots } = data
+    return (
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/components/EventCards/InternalCSEventCard/InternalCSEventCard.tsx:136]: Shall we add proper alias for objectsnapshots[0].value  and objectsnapshots[1].value

Comment (deepesh-ui) [reply src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/components/EventCards/InternalCSEventCard/InternalCSEventCard.tsx:136]: add ?. as well

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/__tests__/ChangeEventCard.mock.ts [hunk 3]
```diff
@@ -182,3 +182,87 @@ export const HarnessFFMockData = {
   },
   responseMessages: []
 }
+
+export const HarnessFFYAMLSnapshots = {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/components/ChangeEventCard/__tests__/ChangeEventCard.mock.ts:186]: rename this HarnessFFYAMLResponse or something suitable

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger coverage

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger coverage

**Review by deepesh-ui — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**